syntax = "proto3";

package wallet;

service Wallet {
  rpc balance(BalanceRequest) returns (BalanceReply) {}
  rpc create_payment_order(CreatePaymentOrderRequest)
      returns (CreatePaymentOrderReply) {}
  rpc transfer(TransferRequest) returns (TransferReply) {}
  rpc end_execution(EndExecutionRequest) returns (EndExecutionReply) {}
}

message BalanceRequest { string wallet = 1; }

message BalanceReply { int32 value = 1; }

message CreatePaymentOrderRequest {
  string wallet = 1;
  int32 value = 2;
}

message CreatePaymentOrderReply { int32 retval = 1; }

message TransferRequest {
  int32 payment_order = 1;
  int32 recount = 2;
  string wallet = 3;
}

message TransferReply { int32 status = 1; }

message EndExecutionRequest {}

message EndExecutionReply { int32 pendencies = 1; }
